plugin/src/main/java/net/dzikoysk/funnyguilds/feature/scoreboard/dummy/DummyManager.java

```diff
@@ -1,36 +1,35 @@
 package net.dzikoysk.funnyguilds.feature.scoreboard.dummy;
 
-import net.dzikoysk.funnyguilds.FunnyGuilds;
-import net.dzikoysk.funnyguilds.config.PluginConfiguration;
 import net.dzikoysk.funnyguilds.feature.scoreboard.ScoreboardService;
 import net.dzikoysk.funnyguilds.user.User;
 import net.dzikoysk.funnyguilds.user.UserManager;
 import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import panda.std.Pair;
 import panda.std.stream.PandaStream;
 
 public class DummyManager {
 
-    private final PluginConfiguration pluginConfiguration;
     private final UserManager userManager;
     private final ScoreboardService scoreboardService;
 
-    public DummyManager(FunnyGuilds plugin, ScoreboardService scoreboardService) {
-        this.pluginConfiguration = plugin.getPluginConfiguration();
-        this.userManager = plugin.getUserManager();
+    public DummyManager(UserManager userManager, ScoreboardService scoreboardService) {
+        this.userManager = userManager;
         this.scoreboardService = scoreboardService;
     }
 
     public void updatePlayers() {
         PandaStream.of(Bukkit.getOnlinePlayers())
-                .flatMap(player -> this.userManager.findByUuid(player.getUniqueId()))
-                .forEach(this::updateScore);
+                .mapOpt(player -> this.userManager.findByUuid(player.getUniqueId())
+                        .map(user -> Pair.of(player, user)))
+                .forEach(pair -> this.updateScore(pair.getFirst(), pair.getSecond()));
     }
 
-    public void updateScore(User user) {
+    public void updateScore(Player player, User user) {
         PandaStream.of(Bukkit.getOnlinePlayers())
-                .flatMap(player -> this.userManager.findByUuid(player.getUniqueId()))
+                .flatMap(onlinePlayer -> this.userManager.findByUuid(onlinePlayer.getUniqueId()))
                 .forEach(onlineUser -> {
-                    this.scoreboardService.updatePlayer(onlineUser);
+                    this.scoreboardService.updatePlayer(player, onlineUser);
                     onlineUser.getCache().getDummy().updateScore(user);
                 });
     }
```

Comment:
I think you could use the same logic as in IndividualNameTagManager for updating scores here.